internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -306,6 +307,14 @@ func deployOpenTelemetryCollector(ctx context.Context, rr *odhtypes.Reconciliati
 			Path: CollectorServiceMonitorsTemplate,
 		},
 	}
+
+	if monitoring.Spec.Metrics != nil {
+		template = append(template, odhtypes.TemplateInfo{
+			FS:   resourcesFS,
+			Path: CollectorPrometheusServiceTemplate,
+		})
+	}
```

Comment:
The new Prometheus Service template is only added when `Metrics != nil`, but the `CollectorServiceMonitorsTemplate` (lines 304-308) is added unconditionally. This creates an inconsistency where `ServiceMonitors` might be deployed without their corresponding Service.

```suggestion
	if monitoring.Spec.Metrics != nil {
	    template = append(template, odhtypes.TemplateInfo{
  		    FS:   resourcesFS,
  		    Path: CollectorServiceMonitorsTemplate,
  	    })
		template = append(template, odhtypes.TemplateInfo{
			FS:   resourcesFS,
			Path: CollectorPrometheusServiceTemplate,
		})
	}
```

Or if ServiceMonitors should always be deployed:
```  
  // Add ServiceMonitors (always deployed for collector health monitoring)
  template = append(template, odhtypes.TemplateInfo{
  	FS:   resourcesFS,
  	Path: CollectorServiceMonitorsTemplate,
  })

  // Add Prometheus Service with TLS (only when metrics collection is enabled)
  if monitoring.Spec.Metrics != nil {
  	template = append(template, odhtypes.TemplateInfo{
  		FS:   resourcesFS,
  		Path: CollectorPrometheusServiceTemplate,
  	})
  }
  
 ```

Reply:
The CollectorServiceMonitorsTemplate is always deployed, but it contains conditional logic ({{- if .Metrics }}) that only renders the data-science-prometheus-monitor ServiceMonitor when metrics are enabled. This ensures the ServiceMonitor and its corresponding Service are always deployed together.
I added the suggested comments for more clarity